### PR TITLE
Adding basic KMS resource

### LINKED ISSUE #213: Add support for AWS KMS
AWS pricing: https://aws.amazon.com/kms/pricing/
Terraform resources: `aws_kms*`

### PR DESCRIPTION
**Pull request by sashman**
Fixes #213

Given a key definition:

```
resource "aws_kms_key" "kms" {}
```

Report the basic key usage:

```
 aws_kms_key.kms
  └─ CMK                                                1  months       1.0000       0.0014        1.0000
  Total                                                                              0.0014        1.0000
```

This PR does not cover cumulative costs incurred by yearly key rotation and does not cover key usage.

### CODE REVIEW

File: internal/providers/terraform/aws/kms_key.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+			Unit:            "months",
+			UnitMultiplier:  1,
+			MonthlyQuantity: decimalPtr(decimal.NewFromInt(cmkCount)),
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("awskms"),
+				ProductFamily: strPtr("Encryption Key"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/-KMS-Keys/")},
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/kms_key.go:32]:
```suggestion
					{Key: "locationType", Value: strPtr("AWS Region")},
					{Key: "usagetype", ValueRegex: strPtr("/KMS-Keys/")},
```
This ensures that if/when AWS Outpost support is added for this resource, the filters still return the price we want.

**Review by alikhajeh1 — CHANGES_REQUESTED** [review]
Thanks @sashman! This is good for key storage, what can be done for key usage? Could the `customer_master_key_spec` and `key_usage` params offer clues as to which of the key usage prices apply? if not, we can just show those 5 prices with no costs so users are aware.

**Comment by sashman** [comment]
@alikhajeh1 
> we can just show those 5 prices with no costs so users are aware.

Do you mean like so?
```
  aws_kms_key.kms
  ├─ Customer master key                                1  months         1.0000       0.0014        1.0000
  ├─ Requests                                           -  10k requests   0.0300            -             -
  ├─ Requests (RSA 2048)                                -  10k requests   0.0300            -             -
  ├─ Requests (ECC GenerateDataKeyPair)                 -  10k requests   0.1000            -             -
  ├─ Requests (Asymmetric)                              -  10k requests   0.1500            -             -
  └─ Requests (RSA GenerateDataKeyPair)                 -  10k requests  12.0000            -             -
  Total                                                                                0.0014        1.0000
```

File: internal/providers/terraform/aws/kms_key.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		"ECC_NIST_P256",
+		"ECC_NIST_P384",
+		"ECC_NIST_P521",
+		"ECC_SECG_P256K1":
+		costComponents = append(costComponents, requestPriceComponent("Requests (Asymmetric)", region, "Asymmetric"))
+		return costComponents
+	}
+
+	costComponents = append(costComponents, requestPriceComponent("Requests", region, "All"))
+	costComponents = append(costComponents, requestPriceComponent("Requests (ECC GenerateDataKeyPair)", region, "GenerateDatakeyPair-ECC"))
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/kms_key.go:65]:
```suggestion
	costComponents = append(costComponents, requestPriceComponent("ECC GenerateDataKeyPair requests", region, "GenerateDatakeyPair-ECC"))
```

this matches the AWS pricing page description

File: internal/providers/terraform/aws/kms_key.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	costComponents = append(costComponents, requestPriceComponent("Requests (RSA GenerateDataKeyPair)", region, "GenerateDatakeyPair-ECC"))
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/kms_key.go:66]:
```suggestion
	costComponents = append(costComponents, requestPriceComponent("RSA GenerateDataKeyPair requests", region, "GenerateDatakeyPair-ECC"))
```

File: internal/providers/terraform/aws/kms_key.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		costComponents = append(costComponents, requestPriceComponent("Requests (RSA 2048)", region, "Asymmetric-RSA_2048"))
+		return costComponents
+	case
+		"RSA_3072",
+		"RSA_4096",
+		"ECC_NIST_P256",
+		"ECC_NIST_P384",
+		"ECC_NIST_P521",
+		"ECC_SECG_P256K1":
+		costComponents = append(costComponents, requestPriceComponent("Requests (Asymmetric)", region, "Asymmetric"))
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/kms_key.go:60]:
```suggestion
		costComponents = append(costComponents, requestPriceComponent("Requests (asymmetric)", region, "Asymmetric"))
```

File: internal/providers/terraform/aws/registry.go [hunk 5]
```diff
@@ -180,6 +181,12 @@ var FreeResources []string = []string{
 	"aws_s3_bucket_policy",
 	"aws_s3_bucket_public_access_block",
 
+	// AWS KMS
+	"aws_kms_alias",
+	"aws_kms_ciphertext",
+	"aws_kms_external_key",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/registry.go:187]: The `aws_kms_external_key` needs the `Customer master key` cost only (no requests). Can you possibly add this as a resource and put the `Customer master key` cost in a function so `aws_kms_external_key` calls that function? See the `elb.go` file for an example of how this was done elsewhere.

File: internal/providers/terraform/aws/registry.go [hunk 6]
```diff
@@ -180,6 +181,12 @@ var FreeResources []string = []string{
 	"aws_s3_bucket_policy",
 	"aws_s3_bucket_public_access_block",
 
+	// AWS KMS
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/registry.go:184]: This group should probably go after `// AWS IAM aws_iam_* resources` as we try to keep these grouped alphabetically - both at the group level and at the line-item level, which you've already done :)

**Review by alikhajeh1 — CHANGES_REQUESTED** [review]
Thanks @sashman - just a few minor changes needed, plz assign it back to me when you'd like to review/merge it.

**Pull request merged**